src/server/api/routers/matchRequest.ts

```diff
@@ -229,173 +234,168 @@ export const matchRequestRouter = createTRPCRouter({
   ),
 
   subscribeToAllRequests: protectedProcedure.subscription(() => {
-    return observable<
-      UserRequest & {
-        mode: "add" | "remove" | "update";
-      }
-    >((emit) => {
-      // todo(gab): Consider using only one since currently the subscription only tells to refetch
-      const onAdd = (data: UserRequest) => {
-        emit.next({
-          ...data,
-          mode: "add",
+    return observable<UserRequest & { mode: "add" | "remove" | "update" }>(
+      (emit) => {
+        // todo(gab): Consider using only one since currently the subscription only tells to refetch
+        const onAdd = (data: UserRequest) => {
+          emit.next({ ...data, mode: "add" });
+        };
+        const onRemove = (data: UserRequest) => {
+          emit.next({ ...data, mode: "remove" });
+        };
+        const onUpdate = (data: UserRequest) => {
+          emit.next({ ...data, mode: "update" });
+        };
+        ee.on("update", onUpdate);
+        ee.on("add", onAdd);
+        ee.on("remove", onRemove);
+        return () => {
+          ee.off("add", onAdd);
+          ee.off("remove", onRemove);
+          ee.off("update", onUpdate);
+        };
+      },
+    );
+  }),
+
+  checkAndProcessAutoMatchIfPossible: protectedProcedure.mutation(
```

Comment:
The prev name seems alot more suitable

Reply:
I would prefer if u stopped using format on save, it makes it very hard to review

Reply:
there shouldn't be any changes to this code anyways cause it should have been correct before, can revert to previous state?

Reply:
Mainly cause there's currently too much logic in the critical section, it's a very bad idea

Reply:
^ refer above comment for context
1) the matching logic whilst unconventional is not hard to logic (as u rightfully mentioned previously)

i abstracted it out hoping to use it for the mutex in manual confirmation but in the end opted for a similar sequence of steps. whilst not refactored to use same function should still aid in readability

Reply:
This change is basically adding untested code for no benefit

Reply:
> ^ refer above comment for context
> 
> 1. the matching logic whilst unconventional is not hard to logic (as u rightfully mentioned previously)
> 
> i abstracted it out hoping to use it for the mutex in manual confirmation but in the end opted for a similar sequence of steps. whilst not refactored to use same function should still aid in readability

idt it improves readability

Reply:
Ok as discussed ill revert it back.
I'm ok with both since correctness is the same